app/components/SocketProvider.js

```diff
@@ -39,164 +46,213 @@ class SocketProvider extends React.Component<Props> {
       collections,
       memberships,
       policies,
+      presence,
+      views,
     } = this.props;
     if (!auth.token) return;
 
     this.socket.on('connect', () => {
+      // immediately  current users token to the websocket backend where it is
+      // verified, if all goes well an 'authenticated' message will be received
+      // in response
       this.socket.emit('authentication', {
         token: auth.token,
       });
+    });
```

Comment:
Note: The majority of changes in this file are whitespace changes as the `on` bindings are no longer nested inside of the `connect` callback handler, this caused the event handlers to be bound multiple times – once for every re-connect.